docs/csharp/programming-guide/classes-and-structs/local-functions.md

```diff
@@ -80,22 +80,39 @@ Like with the method iterator, you can refactor the preceding example and put th
 
 At first glance, local functions and [lambda expressions](../../language-reference/operators/lambda-expressions.md) are very similar. In many cases, the choice between using lambda expressions and local functions is a matter of style and personal preference. However, there are real differences in where you can use one or the other that you should be aware of.
 
-Let's examine the differences between the local function and lambda expression implementations of the factorial algorithm. First the version using a local function:
+Let's examine the differences between the local function and lambda expression implementations of the factorial algorithm. Here's the version using a local function:
 
 :::code language="csharp" source="snippets/local-functions/Program.cs" id="FactorialWithLocal" :::
 
-Contrast that implementation with a version that uses lambda expressions:
+This version uses lambda expressions:
 
 :::code language="csharp" source="snippets/local-functions/Program.cs" id="FactorialWithLambda" :::
 
-The local functions have names. The lambda expressions are anonymous methods that are assigned to variables that are `Func` or `Action` types. When you
-declare a local function, the argument types and return type are part of the function declaration. Instead of being part of the body of the lambda expression, the argument types and return type are part of the lambda expression's variable type declaration. Those two differences may result in clearer code.
+### Naming
 
-Local functions have different rules for definite assignment than lambda expressions. A local function declaration can be referenced from any code location where it is in scope. A lambda expression must be assigned to a delegate variable before it can be accessed (or called through the delegate referencing the lambda expression). Notice that the version using the lambda expression must declare and initialize the lambda expression `nthFactorial` before defining it. Not doing so results in a compile time error for referencing `nthFactorial` before assigning it. These differences mean that recursive algorithms are easier to create using local functions. You can declare and define a local function that calls itself. Lambda expressions must be declared, and assigned a default value before they can be re-assigned to a body that references the same lambda expression.
+Local functions are explicitly named like methods. Lambda expressions are anonymous methods and need to be assigned to variables that are either `Func` or `Action` types. When you declare a local function, the process is like writing a normal method; you declare a return type and a function signature.
```

Comment:
```suggestion
Local functions are explicitly named like methods. Lambda expressions are anonymous methods and need to be assigned to variables that are either `Action` or `Func` types. When you declare a local function, the process is like writing a normal method; you declare a return type and a function signature.
```

Order these by type alphabetically, and most appropriate in terms of complexity - since `Func` has return types other than `void`.